### PR TITLE
my progress to ticket 11 so far

### PR DESCRIPTION
**Pull request by Zipeth1010**

### CODE REVIEW

File: Models/Review_models.js [hunk 1]
```diff
@@ -11,13 +12,40 @@ function getReviewByIdModel(id) {
...
+        queryStr += ` GROUP BY reviews.review_id ORDER BY reviews.${sort_by} ${order};`
+    }
+    if (sort_by && !order){
+        queryStr += ` GROUP BY reviews.review_id ORDER BY reviews.${sort_by} DESC;`
+    }
+    if (!sort_by && order){
+        queryStr += ` GROUP BY reviews.review_id ORDER BY reviews.created_at ${order};`
+    }
+    if (!sort_by && !order){
+        queryStr += ` GROUP BY reviews.review_id ORDER BY reviews.created_at DESC;`
```
Comment (RichardNeat) [inline Models/Review_models.js:42]: We may be able to refactor this to make it a little more DRY. Consider the order of these query strings then you could probably just worry about what there 'is' rather than what there 'isn't'. I.e. no if statements with '!' in.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -317,6 +317,77 @@ describe("GET /api/users", () => {
...
+            review_img_url: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number)
+        })
+        })
+        expect(body.reviews).toBeSortedBy("votes", {
+          descending: false
+        })
+      })
+    })
```
Comment (RichardNeat) [inline __tests__/app.test.js:364]: Could we add one more test to prove it can accept an order query

File: __tests__/app.test.js [hunk 3]
```diff
@@ -317,6 +317,77 @@ describe("GET /api/users", () => {
...
+    test("404: If a valid query is fed but there is no results found", () => {
+      return request(app)
+      .get("/api/reviews?category=valid request")
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe("Category not found")
+      })
+    })
+  })
+
```
Comment (RichardNeat) [inline __tests__/app.test.js:390]: There is one more situation to consider, a valid and existent category query that just so happens to have no reviews associated with it yet. This isn't an error as nothing has gone wrong so we need to separate that logic out from any error handling that tests for an empty response. We may need to check if the category exists.

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Nice one Mark some really good code here and great testing, just a couple more considerations to test for and have a go at refactoring the model a bit to make it more DRY if possible. Have a go and pop in another nchelp pr once that's done :)
